src/Forte.Optimizely.ContentUsage/Api/Services/ContentUsageService.cs

```diff
@@ -37,12 +38,26 @@ public IEnumerable<string> GetPageUrls(EPiServerContentUsage contentUsage)
         if (!string.IsNullOrEmpty(url)) 
             return CheckIsPublished(contentLink) ? new[] { url } : new string[] { };
 
-        var pageLinks = _contentSoftLinkRepository.Load(contentLink, true)
+        var pageUrls = this.SearchForPageUrls(contentLink, contentUsage.LanguageBranch);
+
+        return pageUrls;
+    }
+
+    private IEnumerable<string> SearchForPageUrls(ContentReference contentLink, string languageBranch)
+    {
+        var softLinks = this._contentSoftLinkRepository.Load(contentLink, true);
+        var pageLinks = softLinks
             .Where(softLink => softLink.SoftLinkType == ReferenceType.PageLinkReference)
             .Select(softLink => softLink.OwnerContentLink)
-            .Where(ownerContentLink => ownerContentLink != null && CheckIsPublished(ownerContentLink));
+            .Where(ownerContentLink => ownerContentLink != null && this.CheckIsPublished(ownerContentLink));
 
-        return pageLinks.Select(pageLink => _urlResolver.GetUrl(pageLink, contentUsage.LanguageBranch));
+        var pageUrls = pageLinks.SelectMany(pageLink =>
+        {
+            var contentUrl = this._urlResolver.GetUrl(pageLink, languageBranch);
```

Comment:
Currently we run `GetUrl()` before and inside the recursive method (lines 36, 56). With a small rearrangement I guess we can do the check only in the latter place.

```
public IEnumerable<string> GetPageUrls(EPiServerContentUsage contentUsage)
    {
        var contentLink = contentUsage.ContentLink;
        
        var pageUrls = this.SearchForPageUrls(contentLink, contentUsage.LanguageBranch);

        return pageUrls;
    }

    private IEnumerable<string> SearchForPageUrls(ContentReference contentLink, string languageBranch)
    {
        var url = _urlResolver.GetUrl(contentLink, languageBranch);

        if (!string.IsNullOrEmpty(url)) 
            return CheckIsPublished(contentLink) ? new[] { url } : new string[] { };
        
        var softLinks = this._contentSoftLinkRepository.Load(contentLink, true);
        var pageLinks = softLinks
            .Where(softLink => softLink.SoftLinkType == ReferenceType.PageLinkReference)
            .Select(softLink => softLink.OwnerContentLink)
            .Where(ownerContentLink => ownerContentLink != null && this.CheckIsPublished(ownerContentLink));

        var pageUrls = pageLinks.SelectMany(pageLink => this.SearchForPageUrls(pageLink, languageBranch));
        
        return pageUrls;
    }